### PR TITLE
[ADD] inventory: add res at confirmation doc

### PR DESCRIPTION
**Pull request by brse-odoo**
This PR is an [ADD] to add an `at_confirmation` doc into Odoo's Inventory documentation, nested under the shipping_receiving section, and the `reservation_methods` section, specific to 17.0.

Here's what I pushed up in this one:

- New doc for second of 4 reservation methods docs
- Updated toc-tree and added new image folder for doc

**_Important:_**

- _Please ignore the errors on rst. lines 78-79, as the referenced docs have not been added yet, but will as a part of this PR._
- _Please ignore any conflicts, as those are expected and will be resolved as I continue to push up subsequent PRs._

### Merge order:
Merge this PR **after** [PR #7709](https://github.com/odoo/documentation/pull/7709), and **before** [PR #7886](https://github.com/odoo/documentation/pull/7886).

### CODE REVIEW

**Comment by brse-odoo** [comment]
@odoo/inventory-doc-review This PR is ready for review. Per commit message, please ignore any merge conflicts for the time being. Thank you in advance!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:37]:
```suggestion
Click the green :guilabel:` 📈 (forecast graph)` icon on the product line to reveal the product's
```
You can select a different emoji if you think there's a better one.

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:68]:
```suggestion
Availability` field reads `Available` (in green text), and the numbers in the :guilabel:`Demand` and
```

**Review by hojo-odoo — APPROVED** [review]
Hey @brse-odoo,

Great job on this doc! I just had two comments. You can move it to the next round of review after you take a look.

Thanks,
hojo

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:37]: That one works for me! Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:68]: Suggestion implemented ✅

**Comment by brse-odoo** [comment]
Thank you @hojo-odoo for the quick review!

Alright @odoo/us-doc-review, this PR has passed the peer review round, and is now ready for the final content review round. Please feel free to reach out to me if you have any questions. Thank you!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:57]:
```suggestion
   the :guilabel:`On Hand` and :guilabel:`Incoming` quantity, and subtracting the
   :guilabel:`Outgoing` quantity, as shown below:
```
suggesting the use of GUIs here because they appear on the screen/UI. Same reasoning behind your usage of GUIs for 'Forecasted'

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (StraubCreative) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:7]: Can use an abbreviation declaration at the beginning of the doc after the h1, for SO and others:
```
.. |SO| replace:: :abbr:`SO (Sales Order)`
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:16]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:19]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:33]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:53]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:57]: Good catch. Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:7]: I like to write it out upon first mention in the doc, like sales order (SO), and then abbreviate after the first mention. However, if that's not how I should be doing it let me know. I did add in the |SO| abbreviation declaration at the top though.

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:103]: Yep, you did 👍Will ensure that all the implemented changes carry over to this PR. Thank you!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:48]: Let me look into that one -- as far as I understood, I could always see the link despite availability. I'll do some testing and update accordingly.

**Comment by brse-odoo** [comment]
Thank you @ksc-odoo for the quick review!

**Comment by brse-odoo** [comment]
Before sending to technical review, going to resolve the merge conflicts by adding custom anchors (while implementing review changes) for the docs following this one in the merge order: [PR #7886](https://github.com/odoo/documentation/pull/7886), and [PR #7925](https://github.com/odoo/documentation/pull/7925).

Will update this PR once again to resolve merge conflicts, then send to tech review 👍

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:52]: So, I did some testing and found I am able to see the link with 0 product in stock, and even negative forecasted stock. However, I did find that the availability tooltip changes based on the quantity on hand. So I added this admonition below:

.. note::
   If there is **not** sufficient quantity of stock for the product included in the |SO|, the
   :guilabel:`📈 (forecast graph)` icon is red, instead of green.

   Instead of revealing the reserved number of units for the order, the :guilabel:`Availability`
   tooltip reads :guilabel:`Available`, and reveals the available number of units (e.g., `0 Units`).

Thanks for pointing this out! Suggestion implemented ✅

**Comment by brse-odoo** [comment]
> Before sending to technical review, going to resolve the merge conflicts by adding custom anchors (while implementing review changes) for the docs following this one in the merge order: [PR #7886](https://github.com/odoo/documentation/pull/7886), and [PR #7925](https://github.com/odoo/documentation/pull/7925).
> 
> Will update this PR once again to resolve merge conflicts, then send to tech review 👍

All set ✅

**Comment by brse-odoo** [comment]
Alright @samueljlieber, this PR is ready for your technical review. Thank you!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:104]: If the `:show-toc:` metadata is not being used, then the sub-docs will not be linked at the bottom of `reservation_methods.rst`. In this case, you can link to the sub-docs with a seealso :)
```suggestion
.. seealso::
   :doc:`reservation_methods/at_confirmation`

```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:14]: Suggestion to move this seealso to the bottom of the doc :)
```suggestion
.. seealso::
   :doc:`About reservation methods <../reservation_methods>`

```

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @brse-odoo, awesome job on this PR. I only have two small corrections/suggestions, please see below. Once these are addressed tag me again for a quick look, thank you!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:14]: Suggestion implemented ✅

**Comment by brse-odoo** [comment]
> Hi @brse-odoo, awesome job on this PR. I only have two small corrections/suggestions, please see below. Once these are addressed tag me again for a quick look, thank you!

Hi @samueljlieber, thanks for the quick review! This is ready for another look when you have a chance. Thanks!

**Review by samueljlieber — APPROVED** [review]
Looks good to me @brse-odoo, thank you! I am approving now, but this PR needs it's target updated to `17.0` now that https://github.com/odoo/documentation/pull/7709 is fully merged :). Let me know if you have any questions regarding this!

Thank you, once this PR is rebased on 17.0, feel free to tag ZST.

**Comment by brse-odoo** [comment]
> Looks good to me @brse-odoo, thank you! I am approving now, but this PR needs it's target updated to `17.0` now that #7709 is fully merged :). Let me know if you have any questions regarding this!
> 
> Thank you, once this PR is rebased on 17.0, feel free to tag ZST.

Thank you, @samueljlieber! The target for this PR has been updated to 17.0. @StraubCreative This PR is ready for your final lookover, thank you!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (StraubCreative) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:9]: Not a proper noun.
```suggestion
The *at confirmation reservation method* reserves products **only** when a sales order (SO) is
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (StraubCreative) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:40]: - icon name (see screenshot, revealed in dev tools)
- icon may or may not be green for the user, would remove this status indicator (what you did with the **Note** is ok too 👍 )
- remove reference to the screenshot; the writing should be able to stand on it's own *unless* we're writing in 
this way with an **example** admonition block
- add gui for **Reserved** since it is a line item label on the tooltip
Screenshot 2024-03-14 at 1 23 43 PM


```suggestion
Click the :guilabel:`📈 (area graph)` icon on the product line to reveal the product's
:guilabel:`Availability` tooltip, which reveals the :guilabel:`Reserved` number of units for this order.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (StraubCreative) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:44]:
```suggestion
   :guilabel:`📈 (area graph)` icon is red, instead of green.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (StraubCreative) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:56]: nitpick 😉 
```suggestion
   internal link arrow to view the :guilabel:`Forecasted Report` dashboard.
```

**Review by StraubCreative — APPROVED** [review]
Hi @brse-odoo 
Approving with comments. Merge when ready 👍 
.....
@robodoo delegate=brse-odoo

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:9]: That's how it appears in the UI, but I will take this suggestion and apply it throughout the doc. Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:40]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:44]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:56]: Suggestion implemented ✅

**Comment by brse-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
